featureflags/evaluations/evaluator.py

```diff
@@ -214,29 +214,36 @@ def _evaluate_clause(self, clause: Clause, target: Target) -> bool:
         operator = clause.op.lower()
         type = target.get_type(clause.attribute)
 
-        if type is None:
-            if operator == SEGMENT_MATCH_OPERATOR.lower():
-                log.debug("Clause operator is %s, evaluate on segment",
-                          operator)
-                return self._check_target_in_segment(clause.values, target)
-            log.debug("Attribute type %s is none return false", type)
-            return False
-        log.debug("evaluate clause with object %s operator %s and value %s",
-                  type, operator.upper(), clause.values)
-        if operator == IN_OPERATOR.lower():
-            return type.in_list(clause.values)
-        if operator == EQUAL_OPERATOR.lower():
-            return type.equal(clause.values)
-        if operator == GT_OPERATOR.lower():
-            return type.greater_than(clause.values)
-        if operator == STARTS_WITH_OPERATOR.lower():
-            return type.starts_with(clause.values)
-        if operator == ENDS_WITH_OPERATOR.lower():
-            return type.ends_with(clause.values)
-        if operator == CONTAINS_OPERATOR.lower():
-            return type.contains(clause.values)
-        if operator == EQUAL_SENSITIVE_OPERATOR.lower():
-            return type.equal_sensitive(clause.values)
+        try:
+            if type is None:
+                if operator == SEGMENT_MATCH_OPERATOR.lower():
+                    log.debug("Clause operator is %s, evaluate on segment",
+                              operator)
+                    return self._check_target_in_segment(clause.values, target)
+                log.debug("Attribute type %s is none return false", type)
+                return False
+            log.debug("evaluate clause with object %s "
+                      "operator %s and value %s",
+                      type, operator.upper(), clause.values)
+            if operator == IN_OPERATOR.lower():
+                return type.in_list(clause.values)
+            if operator == EQUAL_OPERATOR.lower():
+                return type.equal(clause.values)
+            if operator == GT_OPERATOR.lower():
+                return type.greater_than(clause.values)
+            if operator == STARTS_WITH_OPERATOR.lower():
+                return type.starts_with(clause.values)
+            if operator == ENDS_WITH_OPERATOR.lower():
+                return type.ends_with(clause.values)
+            if operator == CONTAINS_OPERATOR.lower():
+                return type.contains(clause.values)
+            if operator == EQUAL_SENSITIVE_OPERATOR.lower():
+                return type.equal_sensitive(clause.values)
+        except ValueError:
+            log.debug("couldn't convert %s to type %s", clause.values, type)
+        except Exception as e:
+            log.warning("exception processing clause: "
+                        "values: %s, type: %s, error: %s", clause.values, type, e)
```

Comment:
Good catch.  I wonder instead of catching and logging here, which would mean we continue to evaluate a potentially "corrupt" target that may have intended to match a group rule, could we catch exceptions in the public `xVariation` methods?  We are currently only catching `FlagKindMismatch` there, but if we could catch a top level `Exception` we can effectively early return out of bad state like this and immediately return the default variation.   See: https://github.com/harness/ff-python-server-sdk/blob/b01e53fe1c3fc8822a159b8a809ca29cde27d206/featureflags/client.py#L270

See what you think.

Reply:
Certainly you can continue to catch and log here, but could reraise if handling it in `xVariation` .

Reply:
I guess the logic of catching it here is that at this point it's just this particular rule we can't evaluate, which doesn't necessarily mean any of the other rules are invalid or shouldn't be checked after. It's similar to some of the special cases we have at the top of the function where we exit if the clause is empty or the default case where we return false if we don't recognise the operator. We just say this clause couldn't be checked so count it as not matching, log it and move on. 

The main worry with having it bubble the whole way up is that you could have 3 rules that all evaluate to true, then if you change something on SaaS so the type of one rule is wrong everything could be ignored with the flag returning false all of a sudden.

Reply:
> which doesn't necessarily mean any of the other rules are invalid or shouldn't be checked after

Excellent point, you're absolutely right. 👍 

 OOC, had you encountered only `ValueErrors` here, or also the the `Exception` as well?  Just checking if the `Exception` is for something specific or a general guard.